frontend/src/pages/admin/index.js

```diff
@@ -0,0 +1,110 @@
+import React, { useContext } from "react";
+import {
+	Box,
+	Button,
+	Container,
+	Paper,
+	Table,
+	TableBody,
+	TableCell,
+	TableRow,
+	TableHead,
+	Typography,
+	CircularProgress,
+} from "@material-ui/core";
+import api from "../../helpers/api";
+import { AuthContext } from "../../context/auth";
+
+export default class AdminPage extends React.Component {
+	static contextType = AuthContext;
+
+	constructor(props) {
+		super(props);
+		this.state = {
+			allUsersData: undefined,
+			selected: null
+		};
+	}
+
+	async componentDidMount() {
+		const { idToken, uid } = this.context.authState.user;
+		const allUsersData = (await api.admin.users.getAll(idToken)).data.data.users;
+		this.setState({ allUsersData });
+	}
+
+	handleRoleChange = (e, userId, roles) => {
+		const { idToken } = this.context.authState.user;
+		api.admin.users.roles.update(idToken, userId, roles)
+	}
+
+	handleUserDelete = (e, userId) => {
+		const { idToken } = this.context.authState.user;
+		api.admin.users.delete(idToken, userId);
+	}
+
+	userRow(data) {
+		if (data.customClaims == undefined) data.customClaims = {};
+		data.customClaims = {
+			teacher: data.customClaims.teacher || false,
+			superuser: data.customClaims.superuser || false
+		}
+		return (
+			<TableRow hover onClick={this.handleRowSelect}>
+				<TableCell>{data.email}</TableCell>
+				<TableCell>{data.displayName}</TableCell>
+				<TableCell>{data.uid}</TableCell>
+				<TableCell>
+					<input name="teacher" type="checkbox" checked={data.customClaims.teacher} onChange={(e) => {
+						data.customClaims.teacher = !data.customClaims.teacher;
+						this.setState(this.state);
+						this.handleRoleChange(e, data.uid, data.customClaims);
+					}} />
```

Comment:
I'm not sure what is best practice here. How do I improve the state update?

Reply:
Same, move all of this logic into the handleRoleChange function

Reply:
Yeah, usually you can just shove everything into one function instead of declaring a lambda function like that. But that's just code organisation

More importantly, you can't use setState like that, it shouldn't work. Look at the example at [Using states](https://reactjs.org/docs/state-and-lifecycle.html#do-not-modify-state-directly). You need to do something like `this.setState({someValue: "1234test"})`

Reply:
I think `this.setState(this.state);` will actually force a component rerender due to state manipulation. Anyways its just an admin page and Ive given it a quick test and its functional. @mitcmurph can you make a trello task to do the styling and fixt the footer.